libp2p-identify/src/lib.rs

```diff
@@ -49,31 +49,32 @@ mod structs_proto;
 /// Prototype for an upgrade to the identity protocol.
 #[derive(Debug, Clone)]
 pub struct IdentifyProtocol {
-	pub information: IdentifyInfo,
-}
-
-impl IdentifyProtocol {
-	/// Builds a new `IdentifyProtocol`.
-	#[inline]
-	pub fn new(information: IdentifyInfo) -> IdentifyProtocol {
-		IdentifyProtocol {
-			information
-		}
-	}
+	/// Our public key to report to the remote.
+	pub public_key: Vec<u8>,
+	/// Version of the "global" protocol, eg. `ipfs/1.0.0` or `polkadot/1.0.0`.
+	pub protocol_version: String,
+	/// Name and version of the client. Can be thought as similar to the `User-Agent` header
+	/// of HTTP.
+	pub agent_version: String,
```

Comment:
I missed this in the PR that made this choice, but why `String` and not `Bytes` (or a UTF8-enforcing wrapper around it if that's important)? `Bytes` is strictly superior to `Vec<u8>`, especially for small strings (< 32 bytes) and statically-allocated buffers (like we'd expect to find in `protocols`)

Reply:
The `protobuf` library we use to transmit the messages expects a `String`, so using anything else in the API would require a conversion.

Reply:
Ugh, that'd be nice to fix